src/status_im/wallet/choose_recipient/core.cljs

```diff
@@ -68,18 +65,67 @@
             ethereum/default-transaction-gas))
 
 (re-frame/reg-fx
- :resolve-address
+ ::resolve-address
  (fn [{:keys [registry ens-name cb]}]
    (ens/get-addr registry ens-name cb)))
 
+(re-frame/reg-fx
+ ::resolve-addresses
+ (fn [{:keys [registry ens-names callback]}]
+   ;; resolve all addresses then call the callback function with the array of 
+   ;;addresses as parameter
+   (-> (js/Promise.all
```

Comment:
could you please elaborate on why js/Promise here?

Reply:
@yenda preferred to use promises here so will have to comment on this approach since this isn't my code. I had a different implementation that recursively called `reaolve-addresses` to resolve any ENS addresses but i think the idea was that this was a cleaner, presumably less resource intensive approach.

Reply:
@flexsurfer there can be more than one ens name to resolve so using promises allows to do that in parallel and return once all names have been resolved.